### PR TITLE
refactor: ban and unban commands stability improvements and ban handler fixes for manual bans

### PR DESCRIPTION
**Pull request by pdellaert**
### Problem 

When a mod manual bans a user (not using the banCommand), the banHandler gets executed. This will trigger a modLog message indicating this has happened. This modLog message includes the moderator executing the ban and an embed with the details of who is banned, by whom and why.

The next ban using the banCommand, the bot will execute the banCommand through the createMessageHandler.

When the ban happens through the banCommand, the banHandler is still executed because it is still a ban on the server. And while it should not post anything because it should recognize that the command was executed by the bot, it posts a modLog message as if it is a manual ban by the same moderator of the previous manual ban. However, this modLog message embed does not contain the reason, or the moderator executing the command.

### Analysis

The AuditLog is unreliable in timing, as such, the bot will fetch the last log entry, but that might not be the correct one yet. Instead, the banHandler fetches the pervious one, which still refers to the manual (moderator) ban and it fails to recognize the actual ban is now coming from the bot itself, so it posts a message, but a bad one at that (because it can't match up the AuditLog to the guildBanAdd message).

For manual bans, it seems it is more likely to actually be able to fetch the proper AuditLog entry, but there's also a risk there.

### Fixes

* Ban Handler & Ban Command: Moving makeEmbeds out of main code for readability
* Ban Handler & Ban Command: Fail early on multiple types of failures
* Ban Handler: Initial code change to handle old AuditLog entries
* Ban Handler: fix: checking if executor is part of the exclusion list
* Ban Handler: Introduce retry mechanism to fetch matching Audit Log entry
* Ban Command: fix: bug where if the user is no longer in the server the command causes error
* Unban Command: refactor: change unban to capture errors of unknown users instead of stack traces

## Test Results

Screen Shot 2022-10-11 at 10 36 13 PM
Screen Shot 2022-10-11 at 10 40 41 PM
Screen Shot 2022-10-11 at 10 40 48 PM
Screen Shot 2022-10-11 at 10 40 58 PM
Screen Shot 2022-10-11 at 10 42 08 PM
Screen Shot 2022-10-11 at 10 44 45 PM
Screen Shot 2022-10-11 at 10 44 54 PM
Screen Shot 2022-10-11 at 11 01 14 PM
Screen Shot 2022-10-11 at 11 01 23 PM
Screen Shot 2022-10-11 at 11 18 34 PM
Screen Shot 2022-10-11 at 11 18 43 PM

Above did not reproduce duplicate messages.

## Discord Username
straks#7240

### CODE REVIEW

**Comment by pdellaert** [comment]
Screen Shot 2022-10-12 at 12 01 21 AM

File: src/commands/moderation/ban.ts [hunk 1]
```diff
@@ -81,6 +81,12 @@ const failedBanEmbed = (user: User, error: any) => makeEmbed({
     ],
 });
 
+const unknownUserEmbed = (userId: string) => makeEmbed({
+    title: 'Failed to Ban User - Unknown user',
+    color: Colors.Red,
+    description: `Can not ban an unknown user ${userId}. They are not part of this server.`,
```
Comment (benw202) [inline src/commands/moderation/ban.ts:87]: This isn't always when a user isn't in the server. A mod could have grabbed the messageID by accident etc.

Maybe worth a clarification like 'please check the user ID, if it is correct they are not part of the server' but written better haha!

**Review by benw202 — CHANGES_REQUESTED** [review]
One question, otherwise looking good!

File: src/commands/moderation/ban.ts (hunk 1 above)
Comment (pdellaert) [reply src/commands/moderation/ban.ts]: Done! Thanks for the feedback

**Review by benw202 — APPROVED** [review]

**Pull request merged**